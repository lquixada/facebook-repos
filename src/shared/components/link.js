import React from 'react';
import {NavLink} from 'react-router-dom';
import styled from 'styled-components';
import Octicon from 'react-component-octicons';

export const Link = ({to: repo}) => (
  <Wrapper to={`/${repo.name}`}>
    <Icon />
    {repo.name}
    {' '}
    (<Help title="number of contributors">{repo.contributors_count}</Help>)
  </Wrapper>
);


const Icon = styled(Octicon).attrs({
  name: 'repo'
})`
  margin-right: 1rem;
`;

const Wrapper = styled(NavLink).attrs({
  activeClassName: 'active'
})`
  text-decoration: none;
  display: block;
  padding: .8rem 1rem;
  color: #333;
  font-size: 1.2rem;
  border-left: .2rem solid transparent;

  &:hover {
    text-decoration: none;
    color: #327fc7;
    background-color: #fdfdfd;
  }

  &.active {
    color: #333;
    border-left: .2rem solid #d26911;
    background-color: #f4f4f4;
  }
`;

const Help = styled.span`
  color: #327fc7;
  font-size: 1.1rem;
  border-bottom: 1px dashed #327fc7;
  cursor: help;
`;

