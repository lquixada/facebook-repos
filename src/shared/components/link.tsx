import React from 'react'
import RepoIcon from 'react-octicons/lib/repo'
import {NavLink} from 'react-router-dom'
import styled from 'styled-components'

import {flex, imgUrl} from '../helpers'

interface IProps {
  owner: string
  repoName: string
  contributorsCount: number
  isLoading: boolean
}

export const Link = ({owner, repoName, contributorsCount, isLoading}: IProps) => (
  <Wrapper to={`/r/${owner}/${repoName}`}>
    <Icon />
    <Text>
      {repoName}
      {' '}
      (<Help title='number of contributors'>{contributorsCount}</Help>)
    </Text>
    {isLoading ? <img src={imgUrl('loading.gif')} /> : ''}
  </Wrapper>
)

const Wrapper = styled(NavLink)`
  ${flex.display()}
  ${flex.flow('row', 'nowrap')};
  text-decoration: none;
  padding: .8rem 1rem;
  color: #333;
  font-size: 1.2rem;
  border-left: .2rem solid transparent;

  &:hover {
    text-decoration: none;
    color: #2d72b3;
    background-color: #fdfdfd;
  }

  &.active {
    color: #333;
    border-left: .2rem solid #d26911;
    background-color: #f4f4f4;
  }
`

const Icon = styled(RepoIcon)`
  margin-right: .5rem;
`

const Text = styled.span`
  ${flex('1')}
`

const Help = styled.span`
  color: #2d72b3;
  font-size: 1.1rem;
  border-bottom: 1px dashed #2d72b3;
  cursor: help;
`
