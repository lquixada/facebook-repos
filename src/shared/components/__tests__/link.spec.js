import React from 'react';
import {shallow} from 'enzyme';

import {Link} from '../link';

describe('<Link />', () => {
  it('matches the snapshot', () => {
    const component = shallow(<Link repoName="Facebook Repos" contributorsCount={10} />);
    expect(component).toMatchSnapshot();
  });
});
