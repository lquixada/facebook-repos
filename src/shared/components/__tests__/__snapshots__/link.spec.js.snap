// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> matches the snapshot 1`] = `
<Styled(NavLink)
  to="/r/repo1"
>
  <Styled(Octicon)
    name="repo"
  />
  repo1
   
  (
  <styled.span
    title="number of contributors"
  >
    10
  </styled.span>
  )
</Styled(NavLink)>
`;
